### PR TITLE
Feat/viva ms cases officers

### PR DESCRIPTION
**Pull request by dannil76**
Sorry, but this pr includes nonrelated commits. Please focus on following files:
services/stream-eventbridge-ms/lambdas/streamProcessor.js
services/stream-eventbridge-ms/serverless.yml
services/viva-ms/lambdas/syncOfficers.js
services/viva-ms/serverless.yml

What?
Lambda keeping applicant administrators in sync with what's stored in VIVA

How?
'Listening' on database INSERT or MODIFY event from EventBridge
Fetch applicant officers data from VADA via mypage endpoint
Convert applicant officers data to comply with case administrators data model
Comparing officers data with administrator data stored in case (DynamoDB)
Update case administrator data if not in sync with VIVA officers

How to test
Have nodejs version 12 installed
run yarn --no-lock-file from repo root
cd services/stream-eventbridge-ms
run sls deploy
cd ..
cd viva-ms
run sls deploy

Do PUT or POST request on case api to fire off event

Verify that details on case is updated with administrators in DynamoDB

Tested
- [x] My personal AWS environment

### CODE REVIEW

File: services/stream-eventbridge-ms/lambdas/streamProcessor.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const eventsToPut = [];
+
+  for (const record of event.Records) {
+    console.log(`Event: ${record.eventName}/${record.eventID}`);
+
+    const [tableArn] = record.eventSourceARN.split('/stream');
+
+    eventsToPut.push({
+      Time: record.dynamodb.ApproximateCreationDateTime,
+      Source: 'my-service.database',
```
Comment (TeddyAndersson) [inline services/stream-eventbridge-ms/lambdas/streamProcessor.js:19]: is this correct? my-service, feels like boilerplate code :D

File: services/viva-ms/lambdas/syncOfficers.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import hash from '../../../libs/helperHashEncode';
+import * as request from '../../../libs/request';
+import * as dynamoDb from '../../../libs/dynamoDb';
+
+const SSMParams = params.read(config.vada.envsKeyName);
+
+const dynamoDbConverter = AWS.DynamoDB.Converter;
+
+/**
+ * Handler reacting on case event
```
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/syncOfficers.js:18]: I would like a better function description, what's the reason we react on a case event?  To do what?

File: services/viva-ms/lambdas/syncOfficers.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ */
+export const main = async event => {
+  if (event.detail.dynamodb.NewImage === undefined) {
+    // Something is missing... Should not happend
```
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/syncOfficers.js:22]: This comment feels a bit strange, please update

File: services/viva-ms/lambdas/syncOfficers.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      // Something happened in setting up the request that triggered an Error
+      throwError(500, error.message);
+    }
+  }
+
+  return vadaMyPagesResponse.data.person.cases.vivacases.vivacase.officers;
+}
+
+/**
+ * Handler updateing data stored in DynamoDB
```
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/syncOfficers.js:118]: Grammatic error, updateing => updating

File: services/viva-ms/serverless.yml [hunk 5]
```diff
@@ -2,7 +2,7 @@ service: viva-ms
 
 plugins:
   - serverless-bundle
-  - serverless-offline
```
Comment (TeddyAndersson) [inline services/viva-ms/serverless.yml:5]: Don't we need the offline plugin in this ms?

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]
Left some comments for changes in the code due to readability :)

File: services/stream-eventbridge-ms/lambdas/streamProcessor.js (hunk 1 above)
Comment (dannil76) [reply services/stream-eventbridge-ms/lambdas/streamProcessor.js:19]: Correct!

File: services/viva-ms/serverless.yml (hunk 5 above)
Comment (dannil76) [reply services/viva-ms/serverless.yml:5]: No. Can't run event offline?

File: services/viva-ms/lambdas/syncOfficers.js (hunk 4 above)
Comment (TeddyAndersson) [reply services/viva-ms/lambdas/syncOfficers.js:118]: Also what data do we update? Is this a specific function for the case table in dynamodb or is it a general update function for a dynamo table :)

**Comment by TeddyAndersson** [comment]
Tested on my personal AWS, and works as expected (Y)

**Comment by TeddyAndersson** [comment]
Fix or comment requested changes in order to get approval :)

File: services/viva-ms/serverless.yml (hunk 5 above)
Comment (TeddyAndersson) [reply services/viva-ms/serverless.yml:5]: But you can invoke the function if you would like to?

Comment (dannil76) [reply services/viva-ms/serverless.yml:5]: Hmm...

**Review by TeddyAndersson — APPROVED** [review]

**Review by zilaei — APPROVED** [review]
Tested with the app and we can see that the officers are fetched and stored in DB. Good job!

I have some problems in the app starting a form but I do not think that's because of the new officer fields in case.

**Pull request merged**